mbuild/conversion.py

```diff
@@ -842,6 +863,59 @@ def from_rdkit(rdkit_mol, compound=None, coords_only=False, smiles_seed=0):
     return comp
 
 
+def from_gmso(topology, compound=None, coords_only=False, infer_hierarchy=True):
+    """Convert a GMSO Topology to mBuild Compound.
+
+    Parameter
+    ---------
+    topology : gmso.Topology
+        The GMSO Topology to be converted.
+    compound : mb.Compound, optional, default=None
+        Host mb.Compound that we are loading to.
+    coords_only : bool, optional, default=False
+        Set preexisting atoms in compound to coordinates given by Topology.
+    infer_hierarchy : bool, optional, default=True
+        If True, infer compound hierarchy from Topology residue, to be implemented.
+
+    Returns
+    -------
+    compound : mb.Compound
+    """
+    import unyt as u
+
+    if compound and coords_only:
+        if topology.n_sites != compound.n_particles:
+            raise ValueError(
+                f"Number of sites in {topology} does not match {compound}"
+                f"Topology: {topology.n_sites} sites"
+                f"Compound: {compound.n_particles} particles"
+            )
+        atoms_particles = zip(
+            topology.sites, compound.particles(include_ports=False)
+        )
+        if None in compound._particles(include_ports=None):
```

Comment:
Should `include_ports` be a boolean?
```suggestion
        if None in compound._particles(include_ports=False):
```